v1/plugins/logs/plugin.go

```diff
@@ -372,13 +378,34 @@ func (c *Config) validateAndInjectDefaults(services []string, pluginsList []stri
 		return errors.New("invalid decision_log config, specify either 'buffer_size_limit_bytes' or 'max_decisions_per_second'")
 	}
```

Comment:
Should this error condition only be valid for `.Reporting.BufferType == eventBufferType`?

Reply:
So this error is only valid for the `sizeBufferType`, `buffer_size_limit_bytes` is only used by the size buffer. Although this error check should be moved down so that the buffer type and set limits are checked first, updated the PR.